src/circuit/integer/invert.rs

```diff
@@ -0,0 +1,100 @@
+use super::{AssignedCondition, IntegerChip, IntegerInstructions, MainGateInstructions};
+use crate::NUMBER_OF_LIMBS;
+use crate::circuit::{Assigned, AssignedInteger};
+use crate::circuit::main_gate::{CombinationOption, Term};
+use halo2::arithmetic::FieldExt;
+use halo2::circuit::Region;
+use halo2::plonk::Error;
+
+impl<W: FieldExt, N: FieldExt> IntegerChip<W, N> {
+    fn inert_inv_range_tune(&self) -> usize {
+        self.rns.bit_len_prenormalized - (self.rns.bit_len_limb * (NUMBER_OF_LIMBS - 1)) + 1
+    }
+
+    pub(crate) fn _invert(
+        &self,
+        region: &mut Region<'_, N>,
+        a: &AssignedInteger<N>,
+        offset: &mut usize,
+    ) -> Result<(AssignedInteger<N>, AssignedCondition<N>), Error> {
+        let main_gate = self.main_gate();
+
+        let (zero, one) = (N::zero(), N::one());
+        let integer_one = self.rns.new_from_big(1u32.into());
+        // TODO: Shall we just use SynthesisError here?
+        // Passing None to mul is undefined behavior for invert,
+        // throwing an error or panic would be a better choice.
+        let integer_a = a.integer().ok_or(Error::SynthesisError)?;
```

Comment:
In sythesis time value `a.integer()` can be `None`. I think this is not the case if we run the circuit with `MockProver` but proper synthesizer will panic there. Calculation of inversion must only run in prover time while `a.integer()` is not `None`

Reply:
Didn't catch your opinion. I changed the `integer_inv` to be `None` if `a.integer()` is `None`. You may just show me how to correct the code.